scripts/add_fossils.py

```
@@ -314,16 +336,145 @@ def backfill_fossils(data_dir, repo_urls):
+            logger.warning(f"  No snapshots found in {json_file.name}, skipping.")
+            continue
+
+        existing_survivor = existing_fossils.get("survivor", {})
+
+        # 2. Clone or fetch the repo
+        local_repo = temp_dir / repo_name
+        if not local_repo.exists():
+            logger.info(f"  Cloning {repo_url}...")
+            _run_command(["git", "clone", repo_url, str(local_repo)])
+        else:
+            logger.info(f"  Fetching latest...")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)